client/services/SolutionHandlerService.js

```diff
@@ -45,19 +45,28 @@ tie.factory('SolutionHandlerService', [
           // Otherwise, the code doesn't have any obvious syntax errors.
           // Generate a CodeSubmission object that wraps the student's code
           // into a class and appends some test code, then run the whole thing.
+          var correctnessTests = [];
+          var buggyOutputTests = [];
+          var performanceTests = [];
+          for (var i = 0; i < tasks.length; i++) {
+            correctnessTests.push(tasks[i].getCorrectnessTests());
+            buggyOutputTests.push(tasks[i].getBuggyOutputTests());
+            performanceTests.push(tasks[i].getPerformanceTests());
+          }
+          var lastTask = tasks[tasks.length - 1];
           var codeSubmission = CodeSubmissionObjectFactory.create(
             studentCode.trim());
           CodePreprocessorDispatcherService.preprocess(
             language, codeSubmission, auxiliaryCode,
-            task.getInputFunctionName(), task.getMainFunctionName(),
-            task.getOutputFunctionName(), task.getCorrectnessTests(),
-            task.getBuggyOutputTests(), task.getPerformanceTests());
+            lastTask.getInputFunctionName(), lastTask.getMainFunctionName(),
```

Comment:
I am not sure if I understand your question.

So a python script is generated for running. Inside the script the function that student implements should be invoked while testing. Hence the function name is a must. Even through the function name is changed from task to task, we just care about the current one that the student is writing and want to test it on test cases of beginning from the first task to the current.

I don't think we are able to invoke the function that the student implements without the name. 

Do I answer your question?